build.gradle

```diff
@@ -391,8 +391,9 @@ project('spring-integration-http') {
 	description = 'Spring Integration HTTP Support'
 	dependencies {
 		compile project(":spring-integration-core")
-		compile "org.springframework:spring-webmvc:$springVersion"
-		compile "org.springframework:spring-webflux:$springVersion"
+		compile "org.springframework:spring-web:$springVersion"
+		compile ("org.springframework:spring-webmvc:$springVersion", optional)
+		compile ("org.springframework:spring-webflux:$springVersion", optional)
```

Comment:
Perhaps we should split into 3 modules; `http-common`, `http-mvc` (or leave it as `http`) and `http-webflux`, to make the transitive dependencies work ok for users.